nordicsemi/dfu/dfu_transport_ant.py

```diff
@@ -0,0 +1,604 @@
+#
+#   1. Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+#   2. Redistributions in binary form must reproduce the above copyright notice, this
+#   list of conditions and the following disclaimer in the documentation and/or
+#   other materials provided with the distribution.
+#
+#   3. Neither the name of Nordic Semiconductor ASA nor the names of other
+#   contributors to this software may be used to endorse or promote products
+#   derived from this software without specific prior written permission.
+#
+#   4. This software must only be used in or with a processor manufactured by Nordic
+#   Semiconductor ASA, or in or with a processor manufactured by a third party that
+#   is used in combination with a processor manufactured by Nordic Semiconductor.
+#
+#   5. Any software provided in binary or object form under this license must not be
+#   reverse engineered, decompiled, modified and/or disassembled.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# Python imports
+import binascii
+from datetime import datetime, timedelta
+import logging
+import Queue
+import struct
+import sys
+
+
+# Python 3rd party imports
+try:
+    import antlib
+    import antlib.antdefines as antdefines
+    import antlib.antmessage as antmessage
+except ImportError as e:
+    print(e)
+    raise Exception("Try running 'pip install antlib'.")
+
+# Nordic Semiconductor imports
+from nordicsemi.dfu.dfu_transport   import DfuTransport, DfuEvent, TRANSPORT_LOGGING_LEVEL
+from pc_ble_driver_py.exceptions    import NordicSemiException
+
+
+def platform_supported():
+    """
+    A platform check should be performed before using this module.
+    Only a python 32bit interpreter on windows is supported. The 'antlib'
+    package depends on precompiled windows 32bit libraries.
+
+    :return: True if platform is windows 32bit or False otherwise.
+    :rtype: bool
+    """
+    can_run = False
+
+    if sys.platform in ["win32", "win64"]:
+        # Check if Python is 32 bit.
+        if struct.calcsize("P") * 8 == 32:
+            can_run = True
+
+    if not can_run:
+        print("The ant dfu command is only available on Windows with a 32bit Python \n"
```

Comment:
Is the 32 bit limitation due to antlib dependency limitation?

Reply:
That is correct, the 32-bit limitation is due to the antlib dependency.